### ISSUE TITLE
after_commit on update isn't triggered

### ISSUE THREAD

**Issue by ahorek** [rails/rails#30779]
### Steps to reproduce

```ruby
# frozen_string_literal: true

begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"
  #gem "rails", github: "rails/rails"
  #gem "arel", github: "rails/arel"
  gem "rails", '5.1.4'
  gem "sqlite3"
  gem "pg"
  gem "mysql2"
  gem "pry"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
#ActiveRecord::Base.establish_connection(adapter: "postgresql", database: "", username: '', password: "")
#ActiveRecord::Base.establish_connection(adapter: "mysql2", database: "", username: '', password: "")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
    t.string :name
    t.integer :lock_version
  end
end

class Post < ActiveRecord::Base
  after_commit :test_create, :on => :create
  after_commit :test_update, :on => :update
  after_commit :test_destroy, :on => :destroy

  attr_accessor :status

  def test_create
    self.status = 'create'
  end

  def test_update
    self.status = 'update'
  end

  def test_destroy
    self.status = 'destroy'
  end
end

class BugTest < Minitest::Test
  def test_create
    post = Post.create!
    assert_equal 'create', post.status
  end

  def test_update
    post = Post.create!
    post.update_attributes(:name => 'xxx')
    assert_equal 'update', post.status
  end

  def test_destroy
    post = Post.create!
    post.destroy
    assert_equal 'destroy', post.status
  end
end
```

### Expected behavior
after_commit :test_update, :on => :update calls callbacks

### Actual behavior
after_commit :test_update, :on => :update isn't triggered since rails 5.1.0.beta1 and on master, previous versions are fine
the problem is only on models that contain "**lock_version**" column

### System configuration
**Rails version**:
> 5.1.0.beta1

**Ruby version**:
> 2.4.2

**Comment by yatindrarao** [comment]
It's not working after version **5.1.2**. I'd tried `git bisect` and found the problem commit  702ea0b421b8ce9e95565ebaa47dc5fa7648f1af  .

**Comment by chamnap** [comment]
Yeah, I faced this issue on rails 5.1.5 too.

**Comment by aprescott** [comment]
I also just encountered this on Rails 5.1.5. My understanding was that `after_commit` callbacks were intended to run automatically after #18458. Is that no longer true?

**Comment by aprescott** [comment]
Hm, actually, I can't confirm that. Sorry for the noise. I'll delete my comment.

**Comment by bfad** [comment]
This should really be re-opened. One interesting thing I have noticed is that the `after_xxxx_commit` hooks also don't work, but a simple `after_commit` without specifying `on` does work. (This was on Rails 5.1.6)

**Comment by kamipo** [comment]
This should be fixed by #32167.
